python/hail/methods/qc.py

```diff
@@ -139,14 +139,19 @@ def variant_qc(mt, name='variant_qc') -> MatrixTable:
     - `n_het` (``int64``) -- Number of heterozygous samples.
     - `n_non_ref` (``int64``) -- Number of samples with at least one called
       non-reference allele.
-    - `p_hwe` (``float64``) -- Hardy-Weinberg p-value corresponding to
-      the probability that the distribution of genotypes is under Hardy-Weinberg
-      equilibrium. **Assumes that all genotype calls are diploid, and is only
-      defined for biallelic variants**.
-    - `r_expected_het_freq` (``float64``) -- Ratio of heterozygote count to the
-      expected number of heterozygotes under Hardy-Weinberg equilibrium.
-      **Assumes that all genotype calls are diploid, and is only defined for
-      biallelic variants**.
+    - `p_value_hwe` (``float64``) -- p-value for test of Hardy-Weinberg
```

Comment:
this rename seems unnecessary

Reply:
The rename is because `hardy_weingberg_test` ought to parallel the other tests here (and the regressions) in giving a `p_value` and other statistics. So I'd like to just call it `p_value` when called using the function or aggregator, but that's not a good name when its inside the struct created by `variant_qc`. So `p_value_hwe` seems like the right compromise to avoid multiply names for the same thing. Also, I find it clearer than `p_hwe`, where `p` could be probability or something else.

Reply:
the p in p value stands for probability 🤔

Reply:
I just feel uncomfortable breaking people's pipelines for marginal improvements like this

Reply:
No, it was `p_hwe`

Reply:
`p_hwe` is the probability of a less-likely configuration under HWE, right? Is there something that makes p "values" more special than other probabilities? I don't know the standard practices

Reply:
Yes, the p-value for the frequentist test of hardy weinberg equilibrium is the probability of having a less likely number of hets than the number observed given the allele distribution under the null hypothesis of being in equilibrium. The ubiquitous confusion is to read p-values as probabilities of hypotheses (e.g. probability of HWE, which I'd name `p_hwe`) rather than of data under hypothesis. The second reason, though, is to be consistent as the number of frequentist tests proliferate. All the other tests use `p_value` for good reason. I agree that making this change in isolation doesn't reach the level importance of breaking pipelines, but the other changes will necessitate rewriting HWE pipelines regardless. Stepping back, one goal here is to harmonize and clarify for users a bunch of frequentist tests after they built up quasi-independently. That HWE `r_expected_het_freq` was mis-documented for years is concerning.

That said, I definitely see your point about separating the IR-ify changes from the interface changes, and will have a go at creating the IR-only PR now.